units/dim.cpp

```diff
@@ -12,18 +12,20 @@ namespace scipp::units {
 std::unordered_map<std::string, Dim::Id> Dim::builtin_ids{
     {"detector", Dim::Detector},
     {"d-spacing", Dim::DSpacing},
-    {"E", Dim::Energy},
-    {"Delta-E", Dim::EnergyTransfer},
+    {"energy", Dim::Energy},
+    {"energy-transfer", Dim::EnergyTransfer},
     {"event", Dim::Event},
+    {"final-energy", Dim::FinalEnergy},
     {"group", Dim::Group},
+    {"incident-energy", Dim::IncidentEnergy},
```

Comment:
Why have this as a dimension? It's not a single value determined by choppers?

Reply:
Two reasons:
1. Even if there is just a single value, we still need to store it as a coord. We could just use it as a string, but I felt having a named label would make it easier to implement in the unit conversion, since it will be more readily renamable.
2. Multi-Ei experiments, which as far as I understand are very common, even at existing facilites such as ISIS. Our implementation of unit conversion can directly handle this if data is cut up beforehand and given an extra dimension for `Ei`.

Reply:
Ah yes, multi-ei. this approach would also allow us to attach meta-data to that dimension. Such as attrs for chopper settings if needed.